### PR TITLE
24 changes to readme for clarity and more setup

### LINKED ISSUE #24: Changes to README for clarity and more setup
### Description
Update the README for more detailed instructions on running the server / client and also how to add new dependencies. 
* `flask run`, `npm run dev` for running 
* adding new dependencies to either:
  * client: must `cd` into `/src/client` then run `npm install` (with either `--save` or `--save-dev`)  `--save` is for production so things like d3, and other main dependencies used for the site should use `--save`. `--save-dev` is for development (so things like eslint, type checking, and other things that ONLY RUN IN DEVELOPMENT). It might break if not done correctly. (without either tag it doesnt properly add it to the package file so one does need to be used)
  * server: `cd` into `/src/server` then run `source ./venv/bin/activate` then do `pip install (package)` to install it to the env and not the local machine.

### To-do
* [x] Update instructions for running
* [x] Update instructions for adding dependencies

### PR DESCRIPTION
**Pull request by sellsol**
### Description
Update the README for more detailed instructions on running the server / client and also how to add new dependencies. 
* `flask run`, `npm run dev` for running 
* adding new dependencies to either:
  * client: must `cd` into `/src/client` then run `npm install` (with either `--save` or `--save-dev`)  `--save` is for production so things like d3, and other main dependencies used for the site should use `--save`. `--save-dev` is for development (so things like eslint, type checking, and other things that ONLY RUN IN DEVELOPMENT). It might break if not done correctly. (without either tag it doesnt properly add it to the package file so one does need to be used)
  * server: `cd` into `/src/server` then run `source ./venv/bin/activate` then do `pip install (package)` to install it to the env and not the local machine.

### To-do
* [x] Update instructions for running
* [x] Update instructions for adding dependencies

### Notes
Restructured the README for clarity, added some nicer indications for important info to note, and also a little html to make it look nicer / for better navigation.

### Linked issues
Closes #24

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,43 +1,200 @@
...
 
 #### Server
-`cd src/server`
 
-`source ./venv/bin/activate`
+To run the Flask server, run from the root directory:
+
+```
+cd src/server
+flask run
```
Comment (keyman015) [inline README.md:118]: if running the flask server, the command needs to be performed in the venv for the appropriate requirements.

also im not sure if it errors due to the app file being named `server.py` (thats why i added a env variable to the container)

File: README.md [hunk 2]
```diff
@@ -1,43 +1,200 @@
...
+
+Testing frameworks have been set up for project functionality.
+
+#### Server tests
+
+Tests for the Flask server are written with Pytest. It is recommended that you activate your virtual environment first (e.g. venv), then run in the root directory:
+
+```
+cd src/server
+pytest
```
Comment (keyman015) [inline README.md:139]: venv should be activated here as well.

**Review by paulmmaingi — DISMISSED** [review]
LGTM

File: README.md [hunk 3]
```diff
@@ -1,43 +1,200 @@
...
+A script is available to automatically run all the steps listed below for client and server dependencies.
 
-**NOTE: IF USING WSL, THE DOCKER DESKTOP APP MUST BE INSTALLED AND CONFIGURED (https://docs.docker.com/desktop/wsl/)**
+⚠️ The script was built primarily for Linux systems and your system may not be compatible. If it fails (doesn't output "Setup complete") or your OS is incompatible with these instructions, you can install dependencies manually following the sections below. ⚠️
 
-### Run Tests
+The script requires Python Venv to run. If not already installed, run:
+
+```
+sudo apt install python3-venv
```
Comment (keyman015) [inline README.md:48]: can `sudo apt update` be added here as well, as sometimes if someone doesnt like to update it breaks on setup script.

**Review by keyman015 — CHANGES_REQUESTED** [review]
This looks amazing, just some minor comments about venv activation.

File: README.md (hunk 1 above)
Comment (keyman015) [reply README.md:119]: As discussed, this can be a matter of opinion to use venv.

File: README.md (hunk 2 above)
Comment (keyman015) [reply README.md:140]: As above.

**Review by keyman015 — APPROVED** [review]
LGTM

**Pull request merged**